### PR TITLE
Add support for Tfmini+ LiDAR sensor modules with UART interface

### PR DESCRIPTION
**Pull request by RaphDaMan**
## Description: Add support for TFmini Plus Time-Of-Flight (TOF) family single-point short-range LiDAR sensor modules with UART interface.

**Related issue (if applicable):** N/A

## Checklist:
  - [x ] The pull request is done against the latest development branch
  - [x ] Only relevant files were touched
  - [x ] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x ] The code change is tested and works on Tasmota core ESP8266 V.2.7.4.9
  - [x ] The code change is tested and works with core ESP32 V.1.0.5
  - [X ] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

File: tasmota/xsns_88_tfminiplus.ino [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    uint16_t chiptemp = 0;
+} tfminiplus_sensor;
+
+// Software and hardware serial pointers
+#ifdef ESP8266
+TasmotaSerial *TfmpSerial = nullptr;
+#endif  // ESP8266
+#ifdef ESP32
+HardwareSerial *TfmpSerial = nullptr;
+#endif  // ESP32
```
Comment (arendst) [inline tasmota/xsns_88_tfminiplus.ino:100]: No need for this. TasmotaSerial does use hardware serial on ESP32 only.

File: tasmota/xsns_88_tfminiplus.ino [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+            dataReady = addData((char)data);
+            if (dataReady)
+            {
+                TfmpProcessData();
+            }
+        }
+    }
+}
+
+bool addData(char nextChar)
```
Comment (arendst) [inline tasmota/xsns_88_tfminiplus.ino:182]: Use Tfmp prefix for global function addData()

File: tasmota/i18n.h [hunk 3]
```diff
@@ -211,6 +211,8 @@
 #define D_JSON_IMPORT "Import"
 #define D_JSON_EXPORT "Export"
 #define D_JSON_TOTAL_ACTIVE "TotalActive"
+#define D_JSON_SIGNALSTRENGTH "Signal Strength"
+#define D_JSON_CHIPTEMPERATURE "Chip Temperature"
```
Comment (arendst) [inline tasmota/i18n.h:215]: Remove spaces from JSON field names (Tasmota Standard)

File: tasmota/xsns_88_tfminiplus.ino (hunk 1 above)
Comment (RaphDaMan) [reply tasmota/xsns_88_tfminiplus.ino:100]: Ok, fine for me.

File: tasmota/xsns_88_tfminiplus.ino (hunk 2 above)
Comment (RaphDaMan) [reply tasmota/xsns_88_tfminiplus.ino:182]: Sure (was not consequent) and changed to Tfmp prefix. THX

File: tasmota/i18n.h (hunk 3 above)
Comment (RaphDaMan) [reply tasmota/i18n.h:215]: Thanks for your remark and changed to be Tasmota Standard compliant.

**Comment by ascillato** [comment]
Hi, this PR has conflicts with the language files of the actual codebase. Please, update your branch to be in sync with development branch. Thanks.

File: tasmota/xsns_88_tfminiplus.ino [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+\*********************************************************************************************/
+
+bool Xsns88(byte callback_id)
+{
+    bool result = false;
+    switch (callback_id) {
+        case FUNC_INIT:
+            TfmpInit();
+            break;
+        case FUNC_EVERY_SECOND:
```
Comment (arendst) [inline tasmota/xsns_88_tfminiplus.ino:226]: I'll suggest to the presence check 
```
if (TfmpSerial && tfminiplus_sensor.ready)
```
here. This way all but FUNC_INIT are ignored when no tfmini is connected. See ``xsns_85_mpu6886.ino`` how to.

File: tasmota/xsns_88_tfminiplus.ino [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+ * - http://en.benewake.com/product
+ * - https://de.aliexpress.com/item/32852024277.html?spm=a2g0s.9042311.0.0.27424c4d5Edizk
+ * - https://de.aliexpress.com/item/4001076614996.html?spm=a2g0s.9042311.0.0.27424c4d5Edizk
+\*****************************************************************************/
+
+#include <TasmotaSerial.h>
+
+// Define driver ID
+#define XSNS_88 88
+
```
Comment (arendst) [inline tasmota/xsns_88_tfminiplus.ino:72]: Pls change to ``XSNS_86``

Comment (RaphDaMan) [reply tasmota/xsns_88_tfminiplus.ino:72]: Done! ...I should have waited until 88 comes to draw... ;-)

File: tasmota/xsns_88_tfminiplus.ino (hunk 4 above)
Comment (RaphDaMan) [reply tasmota/xsns_88_tfminiplus.ino:226]: Proposal included according example.

**Comment by ascillato** [comment]
This Pull Request has conflicts that must be resolved.

The code change must pass CI tests. Your PR cannot be merged unless tests pass

**Comment by RaphDaMan** [comment]
> This Pull Request has conflicts that must be resolved.
> 
> The code change must pass CI tests. Your PR cannot be merged unless tests pass

Dear ascillato,
hope I solved the merge conflicts now. Otherwise I would need your support..
Cheers,
Raphael

**Comment by ascillato** [comment]
Sorry, but the conflict is still there. You need to rebase your branch to be up to date with the development branch of the upstream of Tasmota.

If you are using github desktop, the option is there to do this.

**Comment by RaphDaMan** [comment]
holy moly, i managed the impossible thanks to ascilliatos' hint.
I did fear to mess up all your code...

File: tasmota/xsns_88_tfminiplus.ino (hunk 4 above)
Comment (RaphDaMan) [reply tasmota/xsns_88_tfminiplus.ino]: The switch function isn't anymore so nice as from your guide line proposal...

**Comment by arendst** [comment]
Thx.

**Pull request merged**